### PR TITLE
Add profile type to user

### PR DESCRIPTION
**Pull request by verdefred**
## Summary
We decided to add a profile_type enum field to the user so we can take care of the flow from Supporter to Talent. This will need to have some intermediary steps like applying, waiting for approval and getting an approval

## Notion link

https://www.notion.so/talentprotocol/Upgrade-to-Talent-Account-02e0b226fffa4d52829bb16b4ace9926

## How to test?

Nothing should really change in this PR. Everything should work the same, it's just the foundation for the next PRs. We need to run the rake task before going forward with the next PRs so we always have the correct data live

### CODE REVIEW

File: db/migrate/20220524112720_add_profile_type_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddProfileTypeToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :profile_type, :string, default: "supporter"
```
Comment (RubenSousaDinis) [inline db/migrate/20220524112720_add_profile_type_to_users.rb:3]: add `null: false`?

File: app/services/supporter/upgrade_to_talent.rb [hunk 2]
```diff
@@ -38,4 +39,12 @@ def copy_information_to_talent(user)
     user.talent.banner_data = user.investor.banner_data
     user.talent.save!
   end
+
+  def update_profile_type(user, applying)
+    if applying
+      user.update(profile_type: "applying")
+    else
+      user.update(profile_type: "talent")
```
Comment (RubenSousaDinis) [inline app/services/supporter/upgrade_to_talent.rb:47]: One liner? 

```suggestion
      user.update(profile_type: applying ? "applying" : "talent")
```

File: db/migrate/20220524112720_add_profile_type_to_users.rb (hunk 1 above)
Comment (verdefred) [reply db/migrate/20220524112720_add_profile_type_to_users.rb:3]: good catch, I'll make the change

File: app/services/supporter/upgrade_to_talent.rb (hunk 2 above)
Comment (verdefred) [reply app/services/supporter/upgrade_to_talent.rb:47]: Nice!

File: lib/tasks/tmp/update_users_profile_type.rake [hunk 3]
```diff
@@ -0,0 +1,11 @@
+namespace :users do
+  task update_profile_type: :environment do
+    puts "updating users"
+    User.find_each do |user|
+      if user.profile_type == "investor" && user.talent.present?
+        user.update(profile_type: "talent")
```
Comment (RubenSousaDinis) [inline lib/tasks/tmp/update_users_profile_type.rake:6]: I think we should fail if something goes wrong here. At least for us to be aware something failed.

```suggestion
        user.update!(profile_type: "talent")
```

File: app/models/user.rb [hunk 4]
```diff
@@ -35,6 +35,14 @@ class User < ApplicationRecord
 
   VALID_ROLES = ["admin", "basic"].freeze
 
+  enum profile_type: {
+    supporter: "supporter",
+    applying: "applying",
+    waiting_for_approval: "waiting_for_approval",
+    approved: "approved",
```
Comment (RubenSousaDinis) [inline app/models/user.rb:42]: what's the difference between "approved" and "talent"?

Comment (verdefred) [reply app/models/user.rb:42]: Approved means someone has approved you so you can launch your token. Talent means you have been approved and have launched your token

Comment (RubenSousaDinis) [reply app/models/user.rb:42]: ah ok! makes sense!

**Review by RubenSousaDinis — APPROVED** [review]
Well done! Nice job!

**Pull request merged**